samples: boards/microbit: Add initial draft of pong game

jhedberg opened the pull request:
Add initial draft of a pong game using the display on the micro:bit
and its two buttons. This draft only supports a solo game where the
ball bounces from the "roof" and lets the player play against himself.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
samples: microbit/pong: Add initial draft of a pong game

Add initial draft of a pong game using the display on the micro:bit
and its two buttons. This draft only supports a solo game where the
ball bounces from the "roof" and lets the player play against himself.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

Commit:
samples: microbit/pong: Add BLE-based multiplayer support

Add support for connecting two micro:bit devices to play pong between
themselves.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>

lpereira commented:
I wouldn't mind a GIF of the game running for documentation purposes.

dbkinder commented on samples/boards/microbit/pong/README.rst:
```diff
@@ -0,0 +1,17 @@
+Overview
+********
+
+Play pong over Bluetooth between two micro:bit devices.
```
Could use more information about the sample.  For example, you should add what you mention in the commit message:

   This pong game using the display on the micro:bit
   and its two buttons. This version currently supports a solo game where the
   ball bounces from the "roof" and lets the player controls both paddles.

You also talk about bluetooth.  How that's involved in this sample?

dbkinder commented on samples/boards/microbit/pong/README.rst:
```diff
@@ -0,0 +1,17 @@
+.. code-block:: console
+
+   $ cd samples/boards/microbit/pong
+   $ make
```
Need a "Running" section to explain how to play the game.  How does it start and end?  Is a score shown?

jhedberg replied:
I extended the overview section with this, since the game is not that complicated (hence not enough text to justify separate section IMO). The text comes as part of the first patch and describes the capabilities of the game once all three patches of this PR have been merged.

dbkinder reviewed (changes requested):
could use more documentation about what the sample does and how it works

Commit:
samples: microbit/pong: Add basic sound support

Add beeps whenever the ball hits the walls or the paddle. Requires a
piezo buzzer connected to pin 0.

Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>
